src/test/java/io/confluent/connect/elasticsearch/helper/ElasticsearchContainer.java

```diff
@@ -218,6 +291,12 @@ protected void configure() {
         .withFileFromClasspath("instances.yml", getFullResourcePath("instances.yml"))
         .withDockerfileFromBuilder(this::buildImage);
 
+    if (isBasicAuthEnabled()) {
+      log.info("Setting up basic authentication in a Docker image");
+      withEnv("ELASTICSEARCH_USERNAME", ELASTIC_SUPERUSER_NAME);
+      withEnv("ELASTIC_PASSWORD", ELASTIC_PASSWORD);
```

Comment:
basicAuth is mutually exclusive
```suggestion
      withEnv("ELASTIC_PASSWORD", ELASTIC_PASSWORD);
      return;
```

Reply:
We can't return here, we have to do `setImage(image);`. But let me restructure the code to make these options mutually exclusive indeed in case some test sets multiple of them.